exchange/auction.go

```diff
@@ -88,61 +111,56 @@ func (a *auction) doCache(ctx context.Context, cache prebid_cache_client.Client,
 	for _, imp := range bidRequest.Imp {
 		expByImp[imp.ID] = imp.Exp
 	}
-	for _, topBidsPerImp := range a.winningBidsByBidder {
-		for _, topBidPerBidder := range topBidsPerImp {
-			impID := topBidPerBidder.bid.ImpID
-			isOverallWinner := a.winningBids[impID] == topBidPerBidder
-			if !includeBidderKeys && !isOverallWinner {
-				continue
+
+	/* bidders vs winners			*/
+	if targData.includeBidderKeys {
+		// cache banners and/or videos found in impsToBiddersTopBids map[string]map[openrtb_ext.BidderName]*pbsOrtbBid
+		for impID := range a.impsToBiddersTopBids {
+			for _, bid_to_cache := range a.impsToBiddersTopBids[impID] {
+				bids_to_cache_arr = append(bids_to_cache_arr, bid_to_cache)
 			}
-			var customCacheKey string
-			var catDur string
-			useCustomCacheKey := false
-			if competitiveExclusion && isOverallWinner {
-				// set custom cache key for winning bid when competitive exclusion applies
-				catDur = bidCategory[topBidPerBidder.bid.ID]
-				if len(catDur) > 0 {
-					customCacheKey = fmt.Sprintf("%s_%s", catDur, hbCacheID)
-					useCustomCacheKey = true
-				}
+		}
+	} else if targData.includeWinners {
+		// cache banners and/or videos found in impsToTopBids map[string]*pbsOrtbBid
+		for _, bid_to_cache := range a.impsToTopBids {
+			bids_to_cache_arr = append(bids_to_cache_arr, bid_to_cache)
+		}
+	}
+
+	for _, bid_to_cache := range bids_to_cache_arr {
+		/* banners vs videos 			*/
+		if targData.includeCacheBids { //banner
+			if jsonBytes, err := json.Marshal(bid_to_cache.bid); err == nil {
```

Comment:
The old code had a check here to detect an error if trying to use custom cache key for a non-vast bid.  Is there a reason the check is gone?

Reply:
Good catch! Error check included again